### PR TITLE
[CDF-22867] 😵‍💫 Function Schedule bug

### PR DESCRIPTION
**Pull request by doctrino**
# Description

### User Error Message
image

## Checklist

- [ ] Tests added/updated.
- [ ] Run Demo Job Locally.
- [ ] Documentation updated.
- [ ] Changelogs updated in [CHANGELOG.cdf-tk.md](https://github.com/cognitedata/toolkit/blob/main/CHANGELOG.cdf-tk.md).
- [ ] Template changelogs updated in [CHANGELOG.templates.md](https://github.com/cognitedata/toolkit/blob/main/CHANGELOG.templates.md).
- [ ] Version bumped.
  [_version.py](https://github.com/cognitedata/toolkit/blob/main/cognite/cognite_toolkit/_version.py) and
  [pyproject.toml](https://github.com/cognitedata/toolkit/blob/main/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite_toolkit/_cdf_tk/loaders/_resource_loaders/function_loaders.py [hunk 1]
```diff
@@ -367,20 +367,17 @@ def retrieve(self, ids: SequenceNotStr[FunctionScheduleID]) -> FunctionSchedules
...
             for schedule in function_schedules:
-                schedule.function_external_id = func.external_id
+                schedule.function_external_id = func_external_id
             schedules.extend(
-                [
-                    schedule
-                    for schedule in function_schedules
-                    if schedule.name in names_by_function[cast(str, func.external_id)]
-                ]
+                [schedule for schedule in function_schedules if schedule.name in names_by_function[func_external_id]]
```
Comment (doctrino) [inline cognite_toolkit/_cdf_tk/loaders/_resource_loaders/function_loaders.py:375]: This is just a minor quality of life improvement.

File: cognite_toolkit/_cdf_tk/loaders/_resource_loaders/function_loaders.py [hunk 2]
```diff
@@ -389,13 +386,13 @@ def create(self, items: FunctionScheduleWriteList) -> FunctionSchedulesList:
...
-                    item,
-                    client_credentials=client_credentials,
-                )
-            )
-        return FunctionSchedulesList(created)
+            created = self.client.functions.schedules.create(item, client_credentials=client_credentials)
+            # The PySDK mutates the input object, such that function_id is set and function_external_id is None.
+            # If we call .get_id on the returned object, it will raise an error we require the function_external_id
+            # to be set.
+            created.function_external_id = id_.function_external_id
```
Comment (doctrino) [inline cognite_toolkit/_cdf_tk/loaders/_resource_loaders/function_loaders.py:393]: This was the bug.

File: tests/test_integration/test_loaders/test_resource_loaders.py [hunk 3]
```diff
@@ -137,6 +140,23 @@ def test_update_function_schedule(
...
+            cron_expression="0 7 * * TUE",
+            function_external_id=dummy_function.external_id,
+            description="This schedule should be ignored as it does not have a function_external_id",
+        )
+        loader = FunctionScheduleLoader(toolkit_client, None)
+
+        created: FunctionSchedulesList | None = None
+        try:
+            created = loader.create(FunctionScheduleWriteList([local]))
+            loader.get_ids(created)
```
Comment (doctrino) [inline tests/test_integration/test_loaders/test_resource_loaders.py:155]: This triggered the bug

File: tests/test_integration/test_loaders/test_resource_loaders.py [hunk 4]
```diff
@@ -91,22 +93,23 @@ def handle(client: CogniteClient, data: dict, function_call_info: dict) -> str:
...
     name = "integration_test_schedule_dummy"
-    if existing_list := cognite_client.functions.schedules.list(
-        function_external_id=dummy_function.external_id, name=name
-    ):
-        created = existing_list[0]
+    if existing_list := cognite_client.functions.schedules.list(name=name):
+        if len(existing_list) > 1:
+            for existing in existing_list[1:]:
+                cognite_client.functions.schedules.delete(existing.id)
+        schedule = existing_list[0]
```
Comment (doctrino) [inline tests/test_integration/test_loaders/test_resource_loaders.py:100]: Some cleanup in the fixture

**Review by ronpal — APPROVED** [review]

**Pull request merged**